### PR TITLE
Use socks instead of tproxy for tor

### PR DESCRIPTION
**Pull request by arj03**
Thanks for the tip @dominictarr

### CODE REVIEW

**Comment by dominictarr** [comment]
looks good, what do I need to do to test this?

File: plugins/onion.js [hunk 1]
```diff
@@ -1,42 +1,82 @@
...
+
+              socket.on('data', function(data) {
+                  onConnection(data = toPull.duplex(data))
+              })
+
+              // Remember to resume the socket stream.
+              socket.resume()
+          })
+          return function () {
+              controlSocket.end()
```
Comment (dominictarr) [inline plugins/onion.js]: race condition here. need to check if `controlSocket` is set yet,
and if not, set a `closed` var, and in the callback `createConnection` close the stream if closed is true.

Comment (arj03) [inline plugins/onion.js]: Thanks. Fixed.

**Comment by arj03** [comment]
For testing see this post:

https://github.com/ssbc/secret-stack/pull/10

**Comment by arj03** [comment]
Also I only converted plugins and not protocol. What is the difference between protocol and plugins? Just so I know how to test it once its converted.

**Pull request merged**